modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/MethodArgumentResolverDelegate.java

```diff
@@ -451,61 +435,25 @@ public String toString() {
 
   private static class TypedValueCursorArgumentResolverDecorator extends ArgumentResolverDecorator {
 
-    private final CursorComponentDecoratorFactory componentDecoratorFactory;
-
-    public TypedValueCursorArgumentResolverDecorator(ArgumentResolver<Object> decoratee,
-                                                     CursorComponentDecoratorFactory componentDecoratorFactory) {
-      super(decoratee);
-      this.componentDecoratorFactory = componentDecoratorFactory;
-    }
-
-    @Override
-    protected Object decorate(Object value, String eventCorrelationId) {
-      return resolveCursor((TypedValue) value,
-                           v -> decorateInput(v, eventCorrelationId, componentDecoratorFactory));
-    }
-  }
-
-  private static class TypedValueArgumentResolverDecorator extends ArgumentResolverDecorator {
-
-    private final CursorComponentDecoratorFactory componentDecoratorFactory;
-
-    public TypedValueArgumentResolverDecorator(ArgumentResolver<Object> decoratee,
-                                               CursorComponentDecoratorFactory componentDecoratorFactory) {
+    public TypedValueCursorArgumentResolverDecorator(ArgumentResolver<Object> decoratee) {
       super(decoratee);
-      this.componentDecoratorFactory = componentDecoratorFactory;
     }
 
     @Override
-    protected Object decorate(Object value, String eventCorrelationId) {
-      Object v = ((TypedValue) value).getValue();
-
-      v = decorateInput(v, eventCorrelationId, componentDecoratorFactory);
-
-      if (v != ((TypedValue) value).getValue()) {
-        return new TypedValue<>(v, DataType.builder()
-            .type(v.getClass())
-            .mediaType(((TypedValue) value).getDataType().getMediaType())
-            .build(), ((TypedValue) value).getByteLength());
-      } else {
-        return value;
-      }
+    protected Object decorate(Object value) {
+      return resolveCursor((TypedValue) value, MessageUtils::decorateInput);
     }
   }
 
   private static class ObjectArgumentResolverDecorator extends ArgumentResolverDecorator {
 
-    private final CursorComponentDecoratorFactory componentDecoratorFactory;
-
-    public ObjectArgumentResolverDecorator(ArgumentResolver<Object> decoratee,
-                                           CursorComponentDecoratorFactory componentDecoratorFactory) {
+    public ObjectArgumentResolverDecorator(ArgumentResolver<Object> decoratee) {
       super(decoratee);
-      this.componentDecoratorFactory = componentDecoratorFactory;
     }
 
     @Override
-    protected Object decorate(Object value, String eventCorrelationId) {
-      return decorateInput(resolveCursor(value), eventCorrelationId, componentDecoratorFactory);
+    protected Object decorate(Object value) {
+      return resolveCursor(value, MessageUtils::decorateInput);
```

Comment:
is this exactly as it was before introducing the the whole decorators thing? This lambda thing seems less performant than what we had in 4.3

Reply:
review all remaining uses of `MessageUtils::decorateInput`. Make sure they're absolutelly necessary. This kind of lambda expressions and stack trace changes are expensive in this context.

Reply:
I can answer that before the whole decorators thing, the `resolveCursor` did not have this variant that takes the additional `UnaryOperator` parameter. When I profiled this I saw some small performance penalty but not really big enough to be concerning. Of course, if we can remove it, that would be ideal.

The thing is, right now it is our way of doing the wrapping with `UnclosableCursorStream` without spreading the logic of `TypedValue` and `CursorProvider` resolution to `DefaultObjectBuilder`, `ParameterGroupObjectBuilder` and `MethodArgumentResolverDelegate`. But maybe we can do this decoration at a different, unique, place?

Reply:
This ties back to a previous comment in terms of being smarter about when to call this method. In any case, please remove the lambda. Lambdas are cool for tasks that execute sporadically. For code that executes constantly and concurrently, it adds unnecessary pressure on the GC and adds a penalty in stack jumps

Reply:
I changed it to no longer use a lambda for this. I created special versions of the cursor resolving methods that have logic specifically for this particular decoration. These should perform better due to inlining and also because, by knowing the exact type of mapping we are doing and the conditions in which it is done, we can be smarter about when and how to create new `TypedValue` wrappings.
The downside is we are mixing decoration logic with cursor resolution. Also, the code is a little bit longer.
I removed the versions of `resolveCursor` that took a mapping function as a parameter (which were created for the ABP support), to avoid the temptation of using them with lambdas, with the performance penalty that could mean.
I will be conducting a few performance tests today.